app/react/V2/Routes/Settings/Account/Account.tsx

```diff
@@ -1,58 +1,73 @@
+/* eslint-disable max-statements */
 /* eslint-disable react/jsx-props-no-spreading */
-import React, { useEffect, useState } from 'react';
+import React, { useRef, useState } from 'react';
 import { IncomingHttpHeaders } from 'http';
-import { RequestParams } from 'app/utils/RequestParams';
-import UsersAPI from 'app/Users/UsersAPI';
-import { UserSchema } from 'shared/types/userType';
-
 import { LoaderFunction, useLoaderData, useRevalidator } from 'react-router-dom';
 import { useForm } from 'react-hook-form';
 import { useSetAtom } from 'jotai';
-import { notificationAtom } from 'app/V2/atoms';
-
-import { InputField } from 'app/V2/Components/Forms';
-import { Button, Card } from 'app/V2/Components/UI';
-import { SettingsContent } from 'app/V2/Components/Layouts/SettingsContent';
+import { ClientUserSchema } from 'app/apiResponseTypes';
+import { FetchResponseError } from 'shared/JSONRequest';
+import { validEmailFormat } from 'V2/shared/formatHelpers';
 import { Translate } from 'app/I18N';
+import { updateUser, getCurrentUser } from 'V2/api/users';
+import { notificationAtom } from 'V2/atoms';
+import { InputField } from 'V2/Components/Forms';
+import { Button, Card, ConfirmationModal } from 'V2/Components/UI';
+import { SettingsContent } from 'V2/Components/Layouts/SettingsContent';
 import { TwoFactorSetup } from './Components/TwoFactorSetup';
 
 const accountLoader =
   (headers?: IncomingHttpHeaders): LoaderFunction =>
   async () =>
-    UsersAPI.currentUser(new RequestParams({}, headers));
+    getCurrentUser(headers);
 
 const Account = () => {
-  const userAccount = useLoaderData() as UserSchema;
+  const userAccount = useLoaderData() as ClientUserSchema;
   const [isSidepanelOpen, setIsSidepanelOpen] = useState(false);
+  const [confirmationModal, setConfirmationModal] = useState(false);
+  const passwordConfirmation = useRef<string>();
+  const formSubmit = useRef<HTMLButtonElement>(null);
   const setNotifications = useSetAtom(notificationAtom);
   const revalidator = useRevalidator();
 
-  type AccountForm = UserSchema & { passwordConfirm?: string };
+  type AccountForm = ClientUserSchema & { passwordConfirm?: string };
+
   const {
     register,
     watch,
     handleSubmit,
-    reset,
-    formState: { errors, isSubmitted },
+    trigger,
+    resetField,
+    formState: { errors, isDirty },
   } = useForm<AccountForm>({
     defaultValues: userAccount,
     mode: 'onSubmit',
   });
 
-  const submit = async (data: AccountForm) => {
+  const onSubmit = async (data: AccountForm, currentPassword: string) => {
     const { passwordConfirm, ...userData } = data;
     userData.password = userData.password ? userData.password : userAccount.password;
-    await UsersAPI.save(new RequestParams(userData));
-    await revalidator.revalidate();
-    setNotifications({
-      type: 'success',
-      text: <Translate>Account updated</Translate>,
-    });
-  };
 
-  useEffect(() => {
```

Comment:
@konzz  I'm not sure why this effect is here, since I can't find a problem with removing it. (Asking you since I think you added this initially)